docs/DeveloperGuide.md

```diff
@@ -446,171 +446,351 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+    * 2a2. Library manager requests information from user.
+    * 2a3. Library manager enters new information.
+      Steps 2a1-2a3 are repeated until the information entered are valid.
 
-    * 2a2. CLM requests information from user.
+  ***Use case resumes from step 3***
 
-    * 2a3. CLM enters new information.
-    
-    Steps 2a1-2a3 are repeated until the information entered are valid.
+* *a. At any time, library manager chooses to cancel the addition of user.<br>
+    * *a1. Library manager clears the command line using the backspace key.<br>
 
-    ***Use case resumes from step 3.***
+  ***Use case ends***
 
-* *a. At any time, CLM chooses to cancel the addition of user.<br>
-
-  * *a1. CLM clears the command line using the backspace key.<br>
-  
-  ***Use case ends.***
 
 #### Use case: UC2 - List all users
-
 #### MSS:
-
-1. CLM intends to list all users.
-2. CLM enters the command.
+1. Library manager requests to list all users.
+2. Library manager enters the command.
 3. MyBookshelf retrieves the information from the Contact List.
 4. MyBookshelf displays a list of all users, including their names, contact information, and any other relevant details.
-5. CLM reviews the list of users.
+5. Library manager reviews the list of users.
 
 ***Use case ends***
 
 #### Extensions:
-* *a. At any time, CLM chooses to cancel the addition of user.<br>
-
-    * *a1. CLM clears the command line using the backspace key.<br>
+* *a. At any time, library manager chooses to cancel to list all users.<br>
+    * *a1. Library manager clears the command line using the backspace key.<br>
 
   ***Use case ends.***
 
+
 #### Use case: UC3 - Edit user's information
+#### MSS:
+1. Library manager requests to edit a user's information.
+2. Library manager enters the command.
+3. MyBookshelf updates the user's information according to the provided changes.
+4. MyBookshelf notifies library manager that the user has been successfully edited.
+
+***Use case ends***
+
+#### Extensions:
+* 2a. MyBookshelf detects invalid index.
+    * 2a1. MyBookshelf notifies library manager with an error message.
+    * 2a2. Library manager <u> finds user (UC4)</u>.
+    * 2a3. If not exist, library manager <u>add user (UC1)</u>, else records down the index for later use in the edit process.
+
+  ***Use case resumes from step 2.***
+
+* 2b. MyBookshelf detects an error in the entered information.
+    * 2b1. MyBookshelf requests for the valid information.
+    * 2b2. Library manager requests information from user.
+    * 2b3. Library manager enters new information.
+      Steps 2b1-2b3 are repeated until the information entered are valid.
+
+  ***Use case resumes from step 3.***
 
+* *a. At any time, library manager chooses to cancel the edition of user.<br>
+    * *a1. Library manager clears the command line using the backspace key.<br>
+
+  ***Use case ends.***
+
+
+#### Use case: UC4 - Find users
 #### MSS:
+1. Library manager requests to find users by keyword.
```

Comment:
I feel **requests** can be changed to **intends**.

Reply:
Changed in new commit.